### DISCUSSION TITLE
Can I create a swift restic repo when I do a Restic_based backup ?

### DISCUSSION THREAD

**Discussion by sulaimanyu** [backube/volsync#1064] [Q&A] [answered]
In the document, there is only a S3 restic repo as an example. 
My question is can I use swift to create the restic repo in volsync backup?

If yes.

how can I setup the env parameters for swift during create volsync secret. 

Thanks

**Comment by tesshuflower** [comment]
@sulaimanyu This should be possible as restic supports it - see the section of the restic docs here: https://restic.readthedocs.io/en/latest/030_preparing_a_new_repo.html#openstack-swift

Essentially you can set those vars in the `restic-config` secret and they should get passed into the mover pod that runs restic.

Env vars from the `restic-config` that get set in the mover pod can be seen here:
https://github.com/backube/volsync/blob/8821a304067383d49c3fdaa99c74be6ad5ba64a1/controllers/mover/restic/mover.go#L337-L387

**Reply by sulaimanyu** [reply]
Thanks very much for your quickly response. 
The swift which I used now is a self-signed certificates https auth url.  is there any config in volsync to ignore this tls cert?
just like parameter '--insecure-tls' in restic repo init: 
restic -r swift:swift-test:/backup snapshots --insecure-tls

**Reply by tesshuflower** [reply, marked as answer]
You can use your own custom certificate authority for this kind of scenario - see the instructions here: https://volsync.readthedocs.io/en/stable/usage/restic/index.html#using-a-custom-certificate-authority

**Reply by sulaimanyu** [reply]
Yes. I just generated a self-signed certs. then it works.
Thank you very much
